src/plugins/data_source_management/public/components/utils.test.ts

```diff
@@ -274,7 +278,26 @@ describe('DataSourceManagement: Utils.ts', () => {
       expect(getDefaultAuthMethod(authenticationMethodRegistery)?.name).toBe(AuthType.NoAuth);
     });
   });
+  describe('handleSetDefaultDatasourceAfterDeletion', () => {
+    test('should remove defaultDataSource setting and set new defaultDataSource if data sources exist', async () => {
+      mockResponseForSavedObjectsCalls(savedObjects.client, 'find', getDataSourcesResponse);
+      mockUiSettingsCalls(uiSettings, 'get', 'test');
 
+      await handleSetDefaultDatasourceAfterDeletion(savedObjects.client, uiSettings);
+      expect(uiSettings.set).toHaveBeenCalled();
```

Comment:
Nit: according to the behavior which set the first data source in the list as default, maybe it's nice to assert that `uiSettings.set` is called with the first data source id. 
```
expect(uiSettings.set).toHaveBeenCalledWith('defaultDataSource', 'test');
```

Reply:
Would do